gokart/task.py

```diff
@@ -373,6 +374,18 @@ def make_unique_id(self) -> str:
             self.task_unique_id = unique_id
         return unique_id
 
+    def to_str_params(self, only_significant=False, only_public=False):
+        if only_significant and (not only_public):
```

Comment:
I don't understand why this condition is necessary.

Reply:
@kitagry This intend to cache only when called from __init__.

There're two solutions
* only cache default parameters: simple impl. , but specific use 
* cache all patterns of parameters: general use, but a little bit comprecated impl.

Reply:
I prefer to name the condition for your intention. For example

```python
init_condition = only_significant and (not only_public)
if !init_condition:
    return super().to_str_params(only_significant, only_public)
```